libs/sdk-ui-filters/src/AttributeFilter/hooks/useAttributeFilterController.ts

```diff
@@ -749,27 +752,33 @@ function useCallbacks(
             if (enableDuplicatedLabelValuesInAttributeFilter) {
                 const displayAsLabel = handler.getDisplayAsLabel();
                 const { attribute } = handlerState;
-                if (!isPrimaryLabelUsed(nextFilter, attribute.data?.displayForms)) {
-                    const primaryDisplayForm = attribute.data?.displayForms.find((df) => df.isPrimary);
-                    if (!primaryDisplayForm) {
-                        throw new Error("No primary display form found.");
-                    }
-                    const primaryLabelRef = primaryDisplayForm.ref;
-                    const filterUsingPrimaryLabel = replaceFilterDisplayForm(nextFilter, primaryLabelRef);
+                if (isPrimaryLabelUsed(nextFilter, attribute.data?.displayForms)) {
```

Comment:
I switched the branches to remove the negation from the condition. Then I added one parameter to the callback in each branch.